pkg/driver/controller_modify_volume.go

```diff
@@ -55,13 +197,27 @@ func (d *controllerService) ModifyVolumeProperties(
 			modifyOptions.VolumeType = value
 		}
 	}
-	if err := d.cloud.ModifyDisk(ctx, name, &modifyOptions); err != nil {
+	modifyVolumeRequest := modifyVolumeRequest{
+		modifyDiskOptions: modifyOptions,
+	}
+	if err := d.addModifyVolumeRequest(ctx, name, &modifyVolumeRequest); err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not modify volume %q: %v", name, err)
+	}
+
+	select {
+	case err := <-d.modifyVolumeManager.requestHandlerMap[name].errChan:
+		d.modifyVolumeManager.deleteRequestHandler(name)
 		return nil, status.Errorf(codes.Internal, "Could not modify volume %q: %v", name, err)
+	case <-d.modifyVolumeManager.requestHandlerMap[name].resultChan:
+		break
 	}
```

Comment:
(see my comment in `startVolumeTimer` for more context)

This should have an extra case for the context expiring.